source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -5,4 +5,90 @@ Access Data From a Cursor
 .. default-domain:: mongodb
 
 Read operations that return multiple documents do not immediately return
-all values matching the query.
+all values matching the query. Because a query can potentially match
+large sets of documents, these operations rely upon an object called a
+:java-docs:`MongoCursor </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCursor.html>`.
+
+A ``MongoCursor`` fetches documents in batches to reduce both memory
+consumption and network bandwidth usage. 
+
+The following functions use cursors internally, and return the results of the operations:
+
+- ``collection.find()``
+- ``collection.aggregate()``
+- ``collection.listIndexes()``
+- ``db.listCollectionNames()``
+
+Append ``.cursor()`` to use these functions as cursors.  
+
+The following shows ``collection.find()`` as a cursor:
+
+.. code-block:: java
+   :copyable: true
+   
+   collection.find().cursor();
+
+Cursor Paradigms
+----------------
+
+You work with cursors using **cursor paradigms**. Most cursor paradigms
+allow you to access query results one document at a time, abstracting
+away network and caching logic. 
+
+.. warning::
+
+   Do not combine different cursor paradigms on a single cursor.
+   Operations such as ``next()`` and ``forEachRemaining()``
+   each predictably modify the original cursor. If you mix these calls
+   on a single cursor, you may receive unexpected results.
+
+.. note::
+
+   When you reach the last result through iteration, the cursor is
+   exhausted which means it ceases to respond to methods 
+   that access the results.
+
+For Each Functional Iteration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Pass a function to the 
+`forEachRemaining() <https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Iterator.html?is-external=true#forEachRemaining(java.util.function.Consumer)>`_ 
+method of any cursor to iterate through results in a functional style:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Cursor.java
+   :language: java
+   :dedent:
+   :start-after: begin forEachRemainingIteration
+   :end-before: end forEachRemainingIteration
+
+Manual Iteration
+~~~~~~~~~~~~~~~~
+
+Use the :java-docs:`hasNext() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCursor.html#hasNext()>`
+method to check if there are any documents available in the cursor, and then use the
+:java-docs:`next() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCursor.html#next()>`
+method to retrieve the next available document from the cursor:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Cursor.java
+   :language: java
+   :dedent:
+   :start-after: begin manualIteration
+   :end-before: end manualIteration
+
+Cursor Utility Methods
```

Comment:
This heading feels out of place. Methods implies there's more than one, but close is the only one in the topic area. Maybe the heading could be "Close Cursors"?

Reply:
I feel that changing it's name wouldn't sound as nice, so I'll make the heading singular

Reply:
How about `Cursor Cleanup` since that is what it does?